evmcore/validation_test.go

```diff
@@ -0,0 +1,774 @@
+package evmcore
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateTx_RejectsNonLegacyTransactions_BeforeEip2718(t *testing.T) {
```

Comment:
nit: first the condition then the effect
```suggestion
func TestValidateTx_BeforeEip2718_RejectsNonLegacyTransactions(t *testing.T) {
```

Reply:
https://github.com/0xsoniclabs/sonic/pull/154/commits/9ea293d2a5d32b4d447845935fa88a7397446828